include/ocpp/common/websocket/websocket_tls_tpm.hpp

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2020 - 2023 Pionix GmbH and Contributors to EVerest
+#ifndef OCPP_WEBSOCKET_TLS_TPM_HPP
+#define OCPP_WEBSOCKET_TLS_TPM_HPP
+
+#include <ocpp/common/evse_security.hpp>
+#include <ocpp/common/websocket/websocket_base.hpp>
+
+#include <queue>
+namespace ocpp {
+
+struct ConnectionData;
+struct WebsocketMessage;
+
+/// @brief Experimental libwebsockets TLS connection
+class WebsocketTlsTPM final : public WebsocketBase {
+public:
+    /// \brief Creates a new Websocket object with the providede \p connection_options
+    explicit WebsocketTlsTPM(const WebsocketConnectionOptions& connection_options,
+                             std::shared_ptr<EvseSecurity> evse_security);
+
+    ~WebsocketTlsTPM();
+
+    void set_connection_options(const WebsocketConnectionOptions& connection_options) override;
+
+    /// \brief connect to a TLS websocket
+    /// \returns true if the websocket is initialized and a connection attempt is made
+    bool connect() override;
+
+    /// \brief Reconnects the websocket using the delay, a reason for this reconnect can be provided with the
+    /// \param reason parameter
```

Comment:
```suggestion
    /// \p reason parameter
```
Instead of declaring `reason parameter`, you wanted to reference the `reason`, right?

Reply:
It was copy/pasted from the websocket_tls.hpp

Reply:
Oh well, oops.

I'm too lazy to render doxygen from these sources right now, this was just a guess.